mithril-aggregator/src/database/provider/certificate.rs

```diff
@@ -339,9 +350,18 @@ impl<'client> Provider<'client> for CertificateRecordProvider<'client> {
     }
 
     fn get_definition(&self, condition: &str) -> String {
-        let aliases = SourceAlias::new(&[("{:certificate:}", "c")]);
+        let aliases = SourceAlias::new(&[("{:certificate:}", "c"), ("{:top_n:}", "top_n")]);
         let projection = Self::Entity::get_projection().expand(aliases);
-        format!("select {projection} from certificate as c where {condition} order by ROWID desc")
+
+        format!(
+            r#"with
+    top_n (max_rowid) as (select max(ROWID) from certificate)
+  select {projection}
+  from certificate as c 
+    cross join top_n
```

Comment:
Maybe the cross join should be done on the `top_n` table to avoid scanning the whole `certificate` table?